src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,198 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(MyLinkedList.Node<T> prev, T element, MyLinkedList.Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        areIndexInSizeRange(index);
+        linkReassignmentForAdd(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        areIndexInRange(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        areIndexInRange(index);
+        final T removedItem = node(index).item;
+        node(index).item = value;
+        return removedItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        areIndexInRange(index);
+        final T removedItem = node(index).item;
+        linkReassignmentForRemove(index);
+        return removedItem;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = findIndex(object);
+        if (index == -1) {
+            return false;
+        }
+        linkReassignmentForRemove(index);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void areIndexInSizeRange(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + " doesn't exist");
+        }
+    }
+
+    private void areIndexInRange(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + " doesn't exist");
+        }
+    }
+
+    private int findIndex(T object) {
+        Node<T> node = first;
+        for (int i = 0; i < size; i++) {
+            if (node.item == object || node.item != null && node.item.equals(object)) {
+                return i;
+            } else {
+                node = node.next;
+            }
+        }
+        return -1;
+    }
+
+    Node<T> node(int index) {
```

Comment:
I would recommend you to move areIndexInRange(index) method inside the method node(index). And in this case you just will receive IndexOutOfBoundsException exception every time you try to call method node(index).
```suggestion
    Node<T> node(int index) {
    areIndexInRange(index);
```

Reply:
Nice one!